demo/3000-home/pages/_app.js

```diff
@@ -1,20 +1,70 @@
-require('@module-federation/nextjs-mf/lib/include-defaults');
+import { useState } from 'react';
+import App from 'next/app';
+import { Layout, version } from 'antd';
+import HostAppMenu from './_menu';
+import SharedNav from '../components/SharedNav';
+import { useMFClient } from '@module-federation/nextjs-mf/lib/client/useMFClient';
 
-// console.log(__webpack_init_sharing__('default'));
-// console.log(__webpack_share_scopes__);
-
-import dynamic from 'next/dynamic';
 import 'antd/dist/antd.css';
 
-const page = import('./_app.real');
-const AppPage = dynamic(() => import('./_app.real'));
-const Page = (props) => {
-  return <AppPage {...props} />;
-};
-Page.getInitialProps = async (ctx) => {
-  const getInitialProps = (await page).default?.getInitialProps;
-  if (getInitialProps) {
-    return getInitialProps(ctx);
-  }
+function MyApp({ Component, pageProps }) {
+  const [MenuComponent, setMenuComponent] = useState(() => HostAppMenu);
+  useMFClient({
```

Comment:
is this just specific this this app/demo?

Reply:
Yep, it's specific for this app.

But I suppose that it will be widely used in many apps, where besides changing main page content, need to change something in Layout (like a left menu) when the remote changes.

Reply:
Gotcha. Just trying to understand scope and requirements on developers to use this new feature